staticjinja/reloader.py

```diff
@@ -42,11 +47,22 @@ def event_handler(self, event_type, src_path):
         if self.should_handle(event_type, src_path):
             print("%s %s" % (event_type, filename))
             if self.site.is_static(filename):
-                files = self.site.get_dependencies(filename)
-                self.site.copy_static(files)
+                self.site.copy_static([filename])
+            elif self.site.is_ignored(filename):
+                return
             else:
-                templates = self.site.get_dependencies(filename)
-                self.site.render_templates(templates)
+                # Here the changed file is a (maybe partial) template or a data
+                # file
+                self.site.dep_graph.update(filename)
+
+                if self.site.is_template(filename):
+                    needs_rendering = [filename]
+                else:
+                    needs_rendering = filter(
+                        self.site.is_template,
+                        self.site.get_dependencies(filename)
+                        )
```

Comment:
As a matter of style, prefer list comprehensions over `filter()` (and `map()` and `reduce()`) since they're considered more readable.